modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java

```diff
@@ -171,6 +179,26 @@ private CsvOutput(Appendable underlying, String newLine, String separator, boole
     this.safeExpressions = safeExpressions;
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Write a header line to the underlying, returning an instance that allows
+   * cells to be written by header name.
+   * <p>
+   * This writes the header line to the underlying.
+   * The returned instance should then be used to write additional rows.
+   *
+   * @param headers  the list of headers
+   * @param alwaysQuote  when true, each column will be quoted, when false, quoting is selective
+   * @return the writer to use from now on
+   * @throws UncheckedIOException if an IO exception occurs
+   */
+  public CsvRowOutputWithHeaders withHeaders(List<String> headers, boolean alwaysQuote) {
```

Comment:
Does this need some documentation saying that it needs to be called before any other outputting method, and should only be called once?
Or can we assume the best from users?